lib/logstash/codecs/avro.rb

```diff
@@ -71,6 +74,13 @@ def decode(data)
     decoder = Avro::IO::BinaryDecoder.new(datum)
     datum_reader = Avro::IO::DatumReader.new(@schema)
     yield LogStash::Event.new(datum_reader.read(decoder))
+  rescue => e
+    if tag_on_failure
+      @logger.error("Avro parse error, original data now in message field", :error => e)
+      yield LogStash::Event.new("message" => data, "tags" => ["_avroparsefailure"])
+    else
+      raise e
+    end
   end
```

Comment:
woops. jumped the gun here. (jarvis merged).

so, the avro code is not that nice. sometimes it returns StandardError, sometimes it fails internally with a NoMethodError on Nil.